description: clang does not support -Wshift-overflow=2 (for arm64 build)
author: Michael Gilbert <mgilbert@debian.org>

--- a/configure.ac
+++ b/configure.ac
@@ -924,7 +924,12 @@ then
         WINE_TRY_CROSSCFLAGS([-Winit-self])
         WINE_TRY_CROSSCFLAGS([-Wpacked-not-aligned],[EXTRACROSSCFLAGS="$EXTRACROSSCFLAGS -Wno-packed-not-aligned"])
         WINE_TRY_CROSSCFLAGS([-Wpragma-pack],[EXTRACROSSCFLAGS="$EXTRACROSSCFLAGS -Wno-pragma-pack"])
-        WINE_TRY_CROSSCFLAGS([-Wshift-overflow=2])
+
+        case $host_cpu in
+          *aarch64*)  WINE_TRY_CFLAGS([-Wshift-overflow]) ;;
+          *) WINE_TRY_CFLAGS([-Wshift-overflow=2]) ;;
+        esac
+
         WINE_TRY_CROSSCFLAGS([-Wstrict-prototypes])
         WINE_TRY_CROSSCFLAGS([-Wtype-limits])
         WINE_TRY_CROSSCFLAGS([-Wunused-but-set-parameter])
@@ -1799,7 +1804,12 @@ then
   WINE_TRY_CFLAGS([-Winit-self])
   WINE_TRY_CFLAGS([-Wpacked-not-aligned],[EXTRACFLAGS="$EXTRACFLAGS -Wno-packed-not-aligned"])
   WINE_TRY_CFLAGS([-Wpragma-pack],[EXTRACFLAGS="$EXTRACFLAGS -Wno-pragma-pack"])
-  WINE_TRY_CFLAGS([-Wshift-overflow=2])
+
+  case $host_cpu in
+    *aarch64*)  WINE_TRY_CFLAGS([-Wshift-overflow]) ;;
+    *) WINE_TRY_CFLAGS([-Wshift-overflow=2]) ;;
+  esac
+
   WINE_TRY_CFLAGS([-Wstrict-prototypes])
   WINE_TRY_CFLAGS([-Wtype-limits])
   WINE_TRY_CFLAGS([-Wunused-but-set-parameter])
